cli/src/cmd_net.rs

```diff
@@ -0,0 +1,1485 @@
+#[command(verbatim_doc_comment)]
+#[command(name = "port")]
+pub struct CmdPort {
+    #[clap(subcommand)]
+    subcmd: PortSubCommand,
+}
+
+#[async_trait]
+impl RunnableCmd for CmdPort {
+    async fn run(&self, ctx: &Context) -> Result<()> {
+        match &self.subcmd {
+            PortSubCommand::Config(cmd) => cmd.run(ctx).await,
+            PortSubCommand::Status(cmd) => cmd.run(ctx).await,
+        }
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+enum PortSubCommand {
+    /// Manage port configuration.
+    Config(CmdPortConfig),
+
+    /// Observe port status.
+    Status(CmdPortStatus),
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "link")]
+pub struct CmdLink {
+    #[clap(subcommand)]
+    subcmd: LinkSubCommand,
+}
+
+#[async_trait]
+impl RunnableCmd for CmdLink {
+    async fn run(&self, ctx: &Context) -> Result<()> {
+        match &self.subcmd {
+            LinkSubCommand::Add(cmd) => cmd.run(ctx).await,
+            LinkSubCommand::Del(cmd) => cmd.run(ctx).await,
+        }
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+enum LinkSubCommand {
+    /// Add a link to a port.
+    Add(CmdLinkAdd),
+
+    /// Remove a link from a port.
+    Del(CmdLinkDel),
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "add")]
+pub struct CmdLinkAdd {
+    /// Id of the rack to add the link to.
+    rack: Uuid,
+
+    /// Switch to add the link to.
+    #[arg(value_enum)]
+    switch: Switch,
+
+    /// Port to add the link to.
+    #[arg(value_enum)]
+    port: Port,
+
+    /// Whether or not to set auto-negotiation
+    #[arg(long)]
+    pub autoneg: bool,
+
+    /// The forward error correction mode of the link.
+    #[arg(value_enum)]
+    pub fec: LinkFec,
+
+    /// Maximum transmission unit for the link.
+    #[arg(long, default_value_t = 1500u16)]
+    pub mtu: u16,
+
+    /// The speed of the link.
+    #[arg(value_enum)]
+    pub speed: LinkSpeed,
+}
+
+#[async_trait]
+impl RunnableCmd for CmdLinkAdd {
+    async fn run(&self, ctx: &Context) -> Result<()> {
+        let mut settings = current_port_settings(ctx, &self.rack, &self.switch, &self.port).await?;
+        let link = LinkConfigCreate {
+            autoneg: self.autoneg,
+            fec: self.fec,
+            mtu: self.mtu,
+            speed: self.speed,
+            //TODO not fully plumbed on the back end yet.
+            lldp: LldpServiceConfigCreate {
+                enabled: false,
+                lldp_config: None,
+            },
+        };
+        match settings.links.get(PHY0) {
+            Some(_) => {
+                return Err(anyhow::anyhow!("only one link per port supported"));
+            }
+            None => {
+                settings.links.insert(String::from(PHY0), link);
+            }
+        }
+        ctx.client()?
+            .networking_switch_port_settings_create()
+            .body(settings)
+            .send()
+            .await?;
+        Ok(())
```

Comment:
do we want any output?

Reply:
For the commands that modify things, I've been using the no news is good news model.